File: Fitness_App/backend/test-ai-models.js
import "dotenv/config";

async function test() {
  const url = `https://generativelanguage.googleapis.com/v1beta/models?key=${process.env.GEMINI_API_KEY}`;
  try {
     const res = await fetch(url);
     const json = await res.json();
     if(json.models) {
        console.log("Models:", json.models.map(m => m.name));
     } else {
        console.log("Response:", json);
     }
  } catch(e) {
     console.error("Crash:", e.message);
  }
}
test();

File: Fitness_App/backend/test_ai_models.py
import os
import httpx
import asyncio
from dotenv import load_dotenv

load_dotenv()

async def test():
    api_key = os.getenv("GEMINI_API_KEY")
    url = f"https://generativelanguage.googleapis.com/v1beta/models?key={api_key}"
    try:
        async with httpx.AsyncClient() as client:
            res = await client.get(url)
            json_data = res.json()
            if 'models' in json_data:
                print("Models:", [m['name'] for m in json_data['models']])
            else:
                print("Response:", json_data)
    except Exception as e:
        print("Crash:", str(e))

if __name__ == "__main__":
    asyncio.run(test())
